pkg/api/norman/customization/clusterregistrationtokens/formatter.go

```diff
@@ -75,6 +78,22 @@ func (f *Formatter) Formatter(request *types.APIContext, resource *types.RawReso
 			token,
 			ca,
 			getWindowsPrefixPathArg(cluster.Spec.RancherKubernetesEngineConfig))
+	} else {
+		// refresh nodeCommand if agent image or env vars changed
+		cluster, err := f.ClusterLister.Get("", clusterID)
+		if err != nil && errors.IsNotFound(err) {
+			cluster, _ = f.Clusters.Get(clusterID, metav1.GetOptions{})
```

Comment:
why not handle this error returned by f.Clusters.Get?

Reply:
I wanted to preserve the old behavior, the UI always showed the node registration command. I didn't want to return an error in case cluster doesn't load. Users might not always set cluster env vars, so would be weird to get an error in that case...if they did, worst case would be getting an old command. 

What do you think?

Reply:
An error can't be returned here anyways, this is inside a formatter. The best that could happen is an error in rancher logs which won't really help a user.

Reply:
So why not log it, it is anyway useful to debug?